src/ai_hawk/linkedIn_easy_applier.py

```diff
@@ -483,7 +485,7 @@ def _handle_upload_fields(self, element: WebElement, job_context: JobContext) ->
                     job_context.job.resume_path = str(self.resume_path.resolve())
                     job_context.job_application.resume_path = str(self.resume_path.resolve())
                     logger.debug(f"Resume uploaded from path: {self.resume_path.resolve()}")
-                else:
+                elif config.LLM_MODEL_TYPE is constants.OPENAI:
                     logger.debug("Resume path not found or invalid, generating new resume")
```

Comment:
Here it should brake execution not debug logs. throw error which can close program

Reply:
@surapuramakhil That would cause the program to exit for anyone not using OpenAI.

It was not my goal to change any functionality. I simply wanted to update the conditional checks to match the current dependency logic. I kept seeing feder's old resume generator throw an error because I'm using Ollama. It was annoying and discracting, so this fix helps get rid of that error message for those who don't need to see it. 

I can add an else statement again with an info log saying that no resume is selected and resume generation is unavailable for the current llm model?

Reply:
this would cause exit - when they are using resume generator and not on open AI - which is clearly what we are as of now. as resume generator only supports open AI.

Reply:
@surapuramakhil no, that is exactly why I opened #820. The logic is incomplete. This program always tries to upload a resume whether the flag is set or not.

Reply:
@Tgenz1213 explain current code behavior

1. No resume is passed 
2. Resume generator won't be called because of LLM type

what would Agent submit? for resume

Reply:
If the flag is set, sure, the program should exit. But I get the error message when running `python main.py`

It's not just showing the prompt, it's trying to upload a resume every single time.

Reply:
> @Tgenz1213 explain current code behavior
> 
> 
> 
> 1. No resume is passed 
> 
> 2. Resume generator won't be called because of LLM type
> 
> 
> 
> what would Agent submit? for resume 

I'm fixing the "call functions with whatever argument we want without handing the error" option. 

Please stop trying to control logic flow with exceptions. 

Back to your two options though, why is there no 3rd option: don't upload anything?

I could understand exit when specifying --resume (should be called --generate-resume to be less confusing) but in my case I'm not using the flag and shouldn't have been seeing that error if in fact we had option 3.

Reply:
Allow me to explain how the logic works along with what I've been talking about. The default behavior we're talking about does not exist. 

Whether --resume is used or not:
1. Currently, if path is not specified, choose resume template
   - This should only appear if the --resume flag is used without an argument
2. When applying to jobs, it checks if path is none or invalid (again, this happens right now whether --resume is used)
   - This condition should be skipped if the flag isn't used
3. If there's no file, it will always try to generate a resume regardless of flags and config - using the resume generator incorrectly caused an unhandled error to be thrown
   - The change in this commit ensures that that the resume generator is being called with the only correct argument (preventing unnecessary errors from appearing)
   - It seems you wish for this to exit the program when users specify --resume. However, the logic must be fixed as I recommended in #820. Once the correct default behavior is implemented, we can re-add logic for --resume (using generator, uploading resumes, and exiting).

Reply:
Did not mean to click resolved.

Reply:
The other option is: make --resume require an argument. 

Let's please think about the basic functionality of a CLI app. No arguments = default behavior. Allowing --resume to have optional arguments is either wrong or misleading. Using --resume alone means it changes the default behavior to do "something." Using --resume <args> changes the behavior of --resume. 

#820 means to explain there's no behavior difference between "python main.py" and "python main.py --resume" when there should be.

Reply:
Okay I've a couple of comments:
1. Since this PR is supposed to be for formatting/styling the whole codebase, this commit should not have been included in this. It should be a separate PR addressing a separate issue. 
2. The `--resume` flag is supposed to require an argument. It is used when you want to use your own resume pdf file instead of having the bot generate a new resume for every application. I've checkout out to this PR, and `--resume` seems to work as expected:
<img width="809" alt="Screenshot 2024-11-17 at 12 50 11 PM" src="https://github.com/user-attachments/assets/8e3674a1-4381-4488-b374-3898b8e4e9ce">
3. In terms of fixing this issue (which should not be included in this PR), this commit will wait until the bot scrolls through jobs, select a job, scroll again, determine suitability, start filling forms and then find out it cant generate resume if you're not using OpenAI api. Instead, a better fix would be to have a check when a user runs `python main.py` and exit the app if they're not using OpenAI api by stating it wont be able to generate resume and they should apply with their own resume file by passing the flag `python main.py --resume /path/to/resume/file`

Reply:
@49Simon I stand corrected, at least for v4.1.0. There's a reason I thought --resume worked without args though. It might be an issue on main and I was keeping that in mind without double checking on this branch. 

I'll revert the commit when I have a chance later and then double check my observations about behavior and other stuff. I'll work this issue and others separately. I just wanted to quickly remedy a bug while I noticed it. Next time I'll open a bug issue or mention it in an existing one.

Reply:
For the resume, I just tested it on main and it still requires an argument. My understanding was it always required one tbh. If you find the instance where it allowed --resume flag without the file path, let me know.

Reply:
Absolutely! I will research this at my next availability.